hw-4/controllers/groups.controller.ts

```diff
@@ -0,0 +1,51 @@
+import { Request, Response } from 'express';
+
+import Group from '@app-models/group';
+import GroupRepository from '../data-access/group.repository';
+import GroupService from '../services/groups.service';
+
+const groupService = new GroupService(new GroupRepository());
+
+export const getGroups = (req: Request, res: Response): void => {
+  groupService.getGroups().then((groups) => res.json(groups));
+};
+
+export const getGroup = (req: Request, res: Response): void => {
+  groupService.getGroupById(req.params.id).then((group) => res.json(group));
+};
+
+export const addGroup = ({ body }: Request, res: Response): void => {
+  const { name, permissions } = body;
+  groupService.add({
+    name,
+    permissions,
+  } as Group)
+    .then((group: Group) => res.json(group))
```

Comment:
We could simply call `.then(res.json)` keeping the `group` variable name free to create an intermediary variable to pass to `add` method.

Reply:
I also thought we can, but unfortunately it causes runtime error (although it compiles).

Reply:
That's curious behavior.